docs/source/app_development/widgets.rst

```diff
@@ -1,106 +1,91 @@
+Widgets and traitlets
+======================
 
-Develop the custom widgets mainly involve two parts:
+Widgets are eventful Python objects that display GUI components, such as a sliders, buttons, progress bars, drowdowns, etc.
+`ipywidgets`_ is a Python package that provides interactive widgets for the use in Jupyter notebooks.
 
-* **Python**: in the Python part, one needs to define the APIs of the widgets.
-  For instance, the color of the widget's background.
-  The traits types should be initialized as:
+An Ipywidget has a parameter, whose value can be ovserved and accessed from Python.
+This is implemented using `traitlets <https://traitlets.readthedocs.io/>`_.
 
+For example, try the following code in a Jupyter notebook:
 
 .. code-block:: python
 
-    from traitlets import Unicode
-
-    color = Unicode('').tag(sync=True)
-
-* **Javascript** (or TypeScript): the javascript part is for the frontend of the widgets.
-  The frontend uses the `backbone.js <https://backbonejs.org/>`_ framework.
-  The communications between the Python and Javascript parts are from the set and get functions of the traits.
-
-.. code-block:: javascript
+    from ipywidgets import FloatSlider
 
-    let color = this.model.get('color');
+    # slider widget that has a `value` traitlet attribute
+    slider = FloatSlider(value=273.0, min=100.0, max=500.0, description="Temperature [K]");
 
-    this.model.set('color', 'red');
+    # Access the slider value as `slider.value`
+    # print(slider.value)
 
-The change of the traits can be monitored and triggle javascript function.
+    def slider_change(change):
+        """Handle slider changes.
 
-.. code-block:: javascript
+        This function is called when the slider value is changed by the user.
+        """
+        print(change["old"])
+        print(change["new"])
+        if change["new"] > 373.0:
+            print("Boiling now!")
 
-    this.model.on('change:color', this._color_change);
+    slider.observe(slider_change, names="value")
 
-Here is a list for some useful custom widgets:
+Creating your own widgets
+=========================
 
-* `widget-periodictable <https://github.com/osscar-org/widget-periodictable>`_ : A interactive periodic table.
-* `widget-bandsplot <https://github.com/osscar-org/widget-bandsplot>`_ : A widget to plot bandstructure and density of states.
-* `widget-jsmol <https://github.com/osscar-org/widget-jsmol>`_ : A widget to use the molecular visualizer Jmol inside the Jupyter.
+Each ipywidget consists of a Python component that defines how to interact with the widget from Python,
+and a Javascript (or TypeScript) component that is responsible for the graphical representation of the widget and communicates updates back to the Jupyter kernel.
 
-*********************
-AiiDAlab base widgets
-*********************
+If your goal is to combine existing widgets into new, reusable components, this can be done in Python without touching the Javascript component.
```

Comment:
```suggestion
If the goal is to combine existing widgets into new, reusable components, this can be done in Python without touching the Javascript component.
```
Please do not directly address the reader. The only exception would be ("see ... ").